src/hsm.cpp

```diff
@@ -104,6 +160,18 @@ openssl::SSL_EVP_PKEY_Ptr HsmEngine::generateKey(const RSASpec &spec,
                                                  const std::string &keyLabel,
                                                  const std::vector<uint8_t> &keyID)
 {
+    try {
+        // We need to make sure that we don't have 2 keys with the same ID.
+        // For that we need to pass empty keyLabel. Otherwise libp11 tries to find
+        // a key with exact keyLabel/keyID combination. This means that libp11 might
+        // not recognize that the key with the same ID is already there.
+        loadPrivateKey(keyID);
+        throw MoCOCrWException("Key with that keyID already exists");
+    } catch (const MoCOCrWException &e) {
+        if (e.what() != std::string("Unable to load private key. Private key not found!")) {
```

Comment:
I am wondering if we shouldn't make this a private static const string.
Somehow I don't have a good feeling checking for the same string, which is used somewhere else, if there is a better option.

Reply:
Done